### PR TITLE
Add deprecation warnings for v7 flags in v6

### PR DESCRIPTION
**Pull request by brophdawg11**
* Add deprecation warnings for v7 future flags in v6
* Add `@deprecated` annotations to `json`/`defer`

### CODE REVIEW

File: .changeset/ninety-queens-thank.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"react-router-dom": minor
+"react-router": minor
+---
+
+Log deprecation warnings for v7 flags
```
Comment (brophdawg11) [inline .changeset/ninety-queens-thank.md:6]:
```suggestion
- Log deprecation warnings for v7 flags
- Add deprecation warnings to `json`/`defer` in favor of returning raw objects
  - These methods will be removed in React Router v7
```

File: .changeset/ninety-queens-thank.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+"react-router-dom": minor
+"react-router": minor
```
Comment (brophdawg11) [inline .changeset/ninety-queens-thank.md:3]:
```suggestion
"react-router": minor
"@remix-run/router": minor
```

**Comment by li-jia-nan** [comment]
How do I turn off these warnings ?

**Comment by pterblgh** [comment]
Hey team, our CI is full of these log lines. What's the recommended approach to disable them? Most packages usually give package consumers and environment variable to turn off logging.
image

These lines are logged 498 times for our repo, probably unnecessarily:
image

What is your recommendation?

**Comment by jkakar** [comment]
How should code like `return json({ error: "some error" }, { status: 400 });` be handled in the new world?

**Comment by brophdawg11** [comment]
We're going to allow you to disable these warnings by setting the flags to `false` in a 6.28.1 release soon: https://github.com/remix-run/react-router/pull/12441

**Pull request merged**